src/visitors/globals.js

```diff
@@ -7,7 +7,7 @@ const VARS = {
     asset.addDependency('process');
     return 'var process = require("process");';
   },
-  global: () => 'var global = (1,eval)("this");',
+  global: () => 'var global = typeof global === "object" ? global : typeof window === "object" ? window :typeof self === "object" ? self : this;',
```

Comment:
Other implementations seem to default to `{}` instead of `this`. Trade-offs / concerns with either approach?

Reply:
In global context `this` always refers to the global object. As we check before with `typeof global === "object"` whether the global object exists, it might be a better solution to use `{}`. I don't feel expert enough to give a definitive answer though.

Reply:
A cosmetic nitpick: there's a space missing in `:typeof`.